dashboard/app/models/user.rb

```diff
@@ -151,6 +153,23 @@ class User < ActiveRecord::Base
   has_many :districts_users, class_name: 'DistrictsUsers'
   has_many :districts, through: :districts_users
 
+  belongs_to :school_info
+  accepts_nested_attributes_for :school_info, reject_if: :check_school_info
+  validates_presence_of :school_info, unless: :school_info_optional?
+
+  def check_school_info(school_info_attr)
```

Comment:
nit: Please document method with YARD syntax. Also, this should be named `check_school_info?`?